process/transactionProcessor.go

```diff
@@ -495,9 +537,46 @@ func (tp *TransactionProcessor) getTxWithSenderAddr(txHash, sender string) (*dat
 	return nil, errors.ErrTransactionNotFound
 }
 
-func (tp *TransactionProcessor) getTxFromObserver(observer *data.NodeData, txHash string) (*data.GetTransactionResponse, bool, bool) {
+func mergeScResultsFromSourceAndDestIfNeeded(
+	sourceTx *data.FullTransaction,
+	destTx *data.FullTransaction,
+	withEvents bool,
+) *data.FullTransaction {
+	if !withEvents {
+		return destTx
+	}
+
+	scResults := append(sourceTx.ScResults, destTx.ScResults...)
+	scResultsNew := removeDuplicatedScResults(scResults)
+
+	destTx.ScResults = scResultsNew
+
+	return destTx
+}
+
+func removeDuplicatedScResults(scResults []*transaction.SmartContractResultApi) []*transaction.SmartContractResultApi {
```

Comment:
Optionally, rename to `getScResultsUnion`, since here you perform a UNION, but not remove anything per se?

Reply:
done.